src/AbstractLexer.php

```diff
@@ -271,10 +272,11 @@ protected function scan($input)
 
         foreach ($matches as $match) {
             // Must remain before 'value' assignment since it can change content
-            $type = $this->getType($match[0]);
+            $firstMatch = $match[0];
```

Comment:
Creating this variable helped make another Psalm error disappear, but I have mixed feelings about that.